Src/WindowsServer/WindowsServer.Shared/Implementation/DataContracts/AzureInstanceComputeMetadata.cs

```diff
@@ -0,0 +1,177 @@
+
+        [DataMember(Name = "vmSize", IsRequired = true)]
+        internal string VmSize { get; set; }
+
+        [DataMember(Name = "subscriptionId", IsRequired = true)]
+        internal string SubscriptionId { get; set; }
+
+        [DataMember(Name = "resourceGroupName", IsRequired = true)]
+        internal string ResourceGroupName { get; set; }
+
+        internal string GetValueForField(string fieldName)
+        {
+            string aimsValue = string.Empty;
+            switch (fieldName.ToLower(CultureInfo.InvariantCulture))
+            {
+                case "ostype":
+                    aimsValue = this.OsType;
+                    break;
+                case "location":
+                    aimsValue = this.Location;
+                    break;
+                case "name":
+                    aimsValue = this.Name;
+                    break;
+                case "offer":
+                    aimsValue = this.Offer;
+                    break;
+                case "platformfaultdomain":
+                    aimsValue = this.PlatformFaultDomain;
+                    break;
+                case "platformupdatedomain":
+                    aimsValue = this.PlatformUpdateDomain;
+                    break;
+                case "publisher":
+                    aimsValue = this.Publisher;
+                    break;
+                case "sku":
+                    aimsValue = this.Sku;
+                    break;
+                case "version":
+                    aimsValue = this.Version;
+                    break;
+                case "vmid":
+                    aimsValue = this.VmId;
+                    break;
+                case "vmsize":
+                    aimsValue = this.VmSize;
+                    break;
+                case "subscriptionid":
+                    aimsValue = this.SubscriptionId;
+                    break;
+                case "resourcegroupname":
+                    aimsValue = this.ResourceGroupName;
+                    break;
+                default:
+                    throw new ArgumentOutOfRangeException(string.Format(CultureInfo.InvariantCulture, "No field named '{0}' in AzureInstanceComputeMetadata.", fieldName));
+            }
+
+            return aimsValue;
+        }
+
+        /// <summary>
+        /// Because the Azure IMS is on a non-routable IP we need to do some due diligence in our accepting
+        /// values returned from it. This method takes the fieldname and value received for that field, and
+        /// if we can test that value against known limitations of that field we do so here. If the test fails
+        /// we return the empty string, otherwise we return the string given.
+        /// </summary>
+        /// <param name="fieldName">Name of the field acquired from the call to Azure IMS.</param>
+        /// <returns>The value of the field, verified, or the empty string.</returns>
+        internal string VerifyExpectedValue(string fieldName)
```

Comment:
please merge this into the above method itself.

Reply:
I respectfully disagree, I believe this to be too much coupling.